File: HackerRank/10 days of Statistics/Day 1 Standard Deviation.py
#!/bin/python3

import math
import os
import random
import re
import sys

#
# Complete the 'stdDev' function below.
#
# The function accepts INTEGER_ARRAY arr as parameter.
#

def stdDev(arr):
    # Print your answers to 1 decimal place within this function
    sum = 0
    result = 0
    for i in range(len(arr)):
        sum += arr[i]
    
    mean = sum / len(arr)
    
    for i in range(len(arr)):
        result += (arr[i] - mean) ** 2
        
    variance = result / len(arr)
    std = math.sqrt(variance)
    
    print("{:.1f}".format(std))
        

if __name__ == '__main__':
    n = int(input().strip())

    vals = list(map(int, input().rstrip().split()))

    stdDev(vals)

File: HackerRank/10 days of Statistics/Day 1 Standard Deviation.cpp
#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>
using namespace std;


int main() {
    /* Enter your code here. Read input from STDIN. Print output to STDOUT */
    int n;
    cin >> n;
    int arr[n], sum = 0;
    double mean = 0.0, result = 0.0, variance = 0.0, stdDeviation = 0.0;
    for(int i = 0; i < n; ++i){
        cin >> arr[i];
        sum += arr[i];
    }
    mean = sum / (n*1.0);
    for(int i = 0; i < n; ++i){
        result += pow((arr[i] - mean), 2);
    }
    variance = result / (n*1.0);
    stdDeviation = sqrt(variance);
    printf("%.1f", stdDeviation);
    return 0;
}
